### ISSUE TITLE
Variable names in .mps files

### ISSUE THREAD

**Issue by rsemenoff** [cvxpy/cvxpy#1857]
The problem is that I want to be able to use the NEOS offline service sometimes. Currently I export the problem to .mps via:
prob.solve(verbose=False,solver="XPRESS",write_mps='hpw15.mps')
However, I don't see any way to correlate the solution values back to the variable names I assigned in creating the problem.

What I'd like to happen is that cvxpy use my variable names in the exported .mps file, OR provide some means to map the variable names in the .mps files to the variable names in the cvxpy.problem

I've considered mapping the artificial names to the assigned variable names by assuming that variables are
exported in the order I created them, but it seems that isnt necessarily the case always.

**Comment by SteveDiamond** [comment]
@maxschaller this is vaguely related to cvxpygen. Maybe some functionality can be added there?

**Comment by maxschaller** [comment]
@rsemenoff are you using the `name` attribute for your cvxpy variables, like in this [example](https://github.com/cvxgrp/cvxpygen/blob/master/examples/main.py#L16)?

**Comment by rsemenoff** [comment]
Yes, that's exactly what I did. I have both array variables and single variables being createdwith the name attribute, and all the variables appear in the mps file as x_someindexnumber.
But I see that exporting problem files is not really supported much in cvxpy, i.e. its only available forthe xpress solver, and even in that case, I still have to deal with the solver running locally and bogging down the local machine.
I like cvxpy because I can make non-linear constraints that more accurately represents the desired solution,but it may be that extra computation that entails is not worth it. The cvxpy models execute reasonably quicklyon neos (unlike on my laptop which takes forever) , but I can't really evaluate the solution quality with the variablesall being obfuscated/aliased. 
    On Wednesday, August 10, 2022 at 12:38:32 PM PDT, Maximilian Schaller ***@***.***> wrote:  
 
 


@rsemenoff are you using the name attribute for your cvxpy variables, like in this example?

**Comment by maxschaller** [comment]
[Apparently](https://github.com/cvxpy/cvxpy/blob/master/cvxpy/reductions/solvers/conic_solvers/xpress_conif.py#L164), xpress stores a variable name for each component of its variable vector. I will look into naming them `yourvariable_0`, `yourvariable_1`, ..., `yourothervariable_0`, etc. - where the suffix denotes the component of the cvxpy variable (inspired by cvxpygen).

**Comment by SteveDiamond** [comment]
This was posted on the cvxpy discord:

Dear CVXPY community, 

We have built some software that automatically generates an API and a frontend for your optimization model from an annotated model file, with CVXPY a supported language and important use case for ourselves personally. If you wish to check it out: https://fuinn.github.io/mos-docs/

Not all CVXPY structures are yet supported, but we are trying to improve all the time. Feel free to try it out (docker images available), we hope it will be useful, and would love to hear any feedback, contact details at link above. Thanks for CVXPY!

**Comment by rsemenoff** [comment]
Maximillian,  If the name parameter could also be a list-like of strings (this is what I have) , would be my preference.  (CPLEX users would also find it easier to transfer their python code over to cvxpy)
Also, Im not aware of any reason to prefer .mps files, just that Neos-Gurobi supports it,but it also supports [AMPL] [GAMS] [LP] [MPS] [NL]
Thanks!Robert
    On Wednesday, August 10, 2022 at 02:16:14 PM PDT, Maximilian Schaller ***@***.***> wrote:  
 
 


Apparently, xpress stores a variable name for each component of its variable vector. I will look into naming them yourvariable_0, yourvariable_1, ..., yourothervariable_0, etc. - where the suffix denotes the component of the cvxpy variable (inspired by cvxpygen).
